vms/platformvm/txs/builder/camino_builder.go

```diff
@@ -490,6 +500,99 @@ func (b *caminoBuilder) NewRegisterNodeTx(
 	return tx, tx.SyntacticVerify(b.ctx)
 }
 
+func (b *caminoBuilder) NewRewardsImportTx() (*txs.Tx, error) {
+	caminoGenesis, err := b.state.CaminoConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	if !caminoGenesis.LockModeBondDeposit {
+		return nil, errWrongLockMode
+	}
+
+	allUTXOsBytes, _, _, err := b.ctx.SharedMemory.Indexed(
+		b.ctx.CChainID,
+		treasury.AddrTraitsBytes,
+		ids.ShortEmpty[:], ids.Empty[:], MaxPageSize,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching atomic UTXOs: %w", err)
+	}
+
+	now := b.clk.Unix()
+
+	utxos := []*avax.UTXO{}
+	for _, utxoBytes := range allUTXOsBytes {
+		utxo := &avax.TimedUTXO{}
+		if _, err := txs.Codec.Unmarshal(utxoBytes, utxo); err != nil {
+			// that means that this could be simple, not-timed utxo
+			continue
+		}
+
+		if utxo.Timestamp <= now-atomic.SharedMemorySyncBound {
+			utxos = append(utxos, &utxo.UTXO)
+		}
+	}
+
+	if len(utxos) == 0 {
+		return nil, errNoUTXOsForImport
```

Comment:
Is this really an error case?

Reply:
it should be error cause that means, cause tx creation should be only called from triggered cross-chain msg handler
and that should only trigger if c-chain said that it exported some rewards